build.gradle

```diff
@@ -106,5 +132,310 @@ subprojects { project ->
+                    }
+
+                    from components.java
+
+                    artifact sourcesJar
+                    artifact javadocJar
+
+                    if (isGrailsPlugin) {
+                        artifact source: "${subproject.buildDir}/classes/groovy/main/META-INF/grails-plugin.xml",
+                                classifier: "plugin",
+                                extension: 'xml'
+                    }
+                    pom.withXml {
+                        def xml = asNode()
+
+                        xml.children().last() + pomInfo
+                        // dependency management shouldn't be included
+                        def n = xml.get("dependencyManagement")
+                        if (n) xml.remove(n)
+                    }
+
+                }
+            }
+        }
+
+        subproject.afterEvaluate {
+            signing {
+                required { isReleaseVersion }
+                sign publishing.publications.maven
+            }
+        }
+
+        tasks.withType(Sign) {
+            onlyIf { isReleaseVersion }
+        }
+
+        //do not generate extra load on Nexus with new staging repository if signing fails
+        tasks.withType(io.github.gradlenexus.publishplugin.InitializeNexusStagingRepository).configureEach {
+            shouldRunAfter(tasks.withType(Sign))
+        }
+    }
+    apply plugin: "org.asciidoctor.jvm.convert"
+
+    configurations {
+        documentation.extendsFrom compileClasspath
+    }
+
+    dependencies {
+        if (configurations.find { it.name == 'documentation' }) {
+            documentation "org.fusesource.jansi:jansi:1.18"
+            documentation "jline:jline:2.14.6"
+            documentation "com.github.javaparser:javaparser-core:3.15.14"
+            documentation "org.apache.groovy:groovy:${groovyVersion}"
+            documentation "org.apache.groovy:groovy-templates:${groovyVersion}"
+            documentation "org.apache.groovy:groovy-ant:$groovyVersion"
+            documentation "org.apache.groovy:groovy-dateutil:$groovyVersion"
+            documentation "org.apache.groovy:groovy-cli-picocli:$groovyVersion"
+        }
+    }
+
+    groovydoc.classpath += configurations.documentation
+}
+
+if (isReleaseVersion) {
+    apply plugin: 'maven-publish'
+    apply plugin: "io.github.gradle-nexus.publish-plugin"
+
+    nexusPublishing {
+        repositories {
+            sonatype {
+                def ossUser = System.getenv("SONATYPE_USERNAME") ?: project.hasProperty("sonatypeOssUsername") ? project.sonatypeOssUsername : ''
+                def ossPass = System.getenv("SONATYPE_PASSWORD") ?: project.hasProperty("sonatypeOssPassword") ? project.sonatypeOssPassword : ''
+                def ossStagingProfileId = System.getenv("SONATYPE_STAGING_PROFILE_ID") ?: project.hasProperty("sonatypeOssStagingProfileId") ? project.sonatypeOssStagingProfileId : ''
```

Comment:
Change `project.hasProperty('x') ? project.x : ''` to `project.findProperty('x') ?: ''`

Reply:
Updated